include/swift/AST/DiagnosticsSema.def

```diff
@@ -4206,6 +4206,13 @@ ERROR(actor_isolated_self_independent_context,none,
         "actor-isolated %0 %1 can not be referenced from an "
         "'@actorIndependent' context",
         (DescriptiveDeclKind, DeclName))
+ERROR(actor_isolated_inout_state,none,
+      "actor-isolated %0 %1 cannot be passed 'inout' to"
+      "%select{| implicitly}2 asynchronous function",
+      (DescriptiveDeclKind, DeclName, bool))
```

Comment:
I'd word this as `cannot be passed 'inout' %select{to asynchronous function|in an implicitly-asynchronous call}`, since it's only viewed as async from outside of actor contexts, and is otherwise synchronous inside of the actor. 

If it's possible, you may also want to emit the explainer Note diagnostic called `actor_isolated_sync_func` on the callee function (if it has a decl), which helps explain why the function is "implicitly asynchronous".

Reply:
Actually, why not just change the wording to be just `cannot be passed 'inout' in an asynchronous call` to cover both cases. Other parts of the typechecker will raise an error if the programmer forgot to mark the call as `await`, making the additional note just extra noise.

Reply:
I agree that specifying `call` probably makes sense since that's what's async. 

I'm a firm believer that if an error occurs due to something that is happening implicitly (even with the await), it should be noted that it is something being done implicitly. This avoids the programmer going back to the functionDecl and saying "hey, wait, I didn't label this async". Something explaining why it is implicitly async would definitely be a good thing to add though.

Reply:
The main reason why I backtracked on emitting the note here in TypeCheckConcurrency is that later on in TypeCheckEffects, we will emit that same note if the programmer forgot to cover that call-site with an `await`, which is a direct indicator that the programmer doesn't know that it's an implicitly async call-site.

I'm fine with emitting that note here in TypeCheckConcurrency for this error too, so feel free to do so if you'd like to. :)

Reply:
Ah, I wasn't quite sure which part you were referring to as being extra noise. If other parts are emitting the explanation on why it's async, that's fine by me, I won't add it here. I'll just add the bit saying that it's the async call.